README.md

```diff
@@ -1,98 +1,94 @@
 ## How to use
 
-The polyfills are built (concatenated & minified) into several bundles that target
-different browsers and spec readiness:
+### Using `webcomponents-bundle.js`
 
-- `webcomponents-hi.js` -- HTML Imports (needed by Safari Tech Preview)
-- `webcomponents-hi-ce.js` -- HTML Imports and Custom Elements v1 (needed by Safari 10)
-- `webcomponents-hi-sd-ce.js` -- HTML Imports, Custom Elements v1 and Shady DOM/CSS (needed by Safari 9, Firefox, Edge)
-- `webcomponents-sd-ce.js` -- Custom Elements and Shady DOM/CSS (no HTML Imports)
-- `webcomponents-lite.js` -- all of the polyfills: HTML Imports, Custom Elements, Shady DOM/CSS and generic platform polyfills (such as ES6 Promise, Constructable events, etc.) (needed by Internet Explorer 11), and Template (needed by IE 11 and Edge)
+The `webcomponents-bundle.js` contains all of the web components polyfills and is
+suitable for use on any supported browser. All of the polyfill code will be loaded
+but each polyfill will only be used based on feature detection.
+The bundle includes Custom Elements, Shady DOM/CSS and generic platform polyfills
+(such as ES6 Promise, Constructable events, etc.) (needed by Internet Explorer 11),
+and Template (needed by IE 11 and Edge).
 
-If you are only targeting a specific browser, you can just use the bundle that's
-needed by it; alternatively, if your server is capable of serving different assets based on user agent, you can send the polyfill bundle that's necessary for the browser making that request.
+The `webcomponents-bundle.js` is very simple to use but it does load code
+that is not needed on most modern browsers, slowing page load. For best performance,
+use the `webcomponents-loader.js`.
 
-## `webcomponents-loader.js`
-Alternatively, this repo also comes with `webcomponents-loader.js`, a client-side
-loader that dynamically loads the minimum polyfill bundle, using feature detection.
-Note that because the bundle will be loaded asynchronously, you should wait for the `WebComponentsReady` before you can safely assume that all the polyfills have
-loaded and are ready to be used (i.e. if you want to dynamically load other custom
-elements, etc.).
+### Using `webcomponents-loader.js`
 
-Additionally, you can check if `window.WebComponents` exists to know if the `WebComponentsReady` event will fire, and you can check if `window.WebComponents.ready` is true to check if the `WebComponentsReady` event has already fired.
+The `webcomponents-loader.js` is a client-side loader that dynamically loads the
+minimum polyfill bundle, using feature detection. Note that because the bundle will be
+loaded asynchronously, scripts that depend on webcomponents APIs *must* be loaded
+using `WebComponents.waitFor(loadCallback)`, for example custom elements definitions. The `loadCallback` function should load scripts that need the polyfills (typically via `import('my-script.js')`) and
+should return a promise that resolves when all scripts have loaded.
 
 Here's an example:
```

Comment:
Let's add an example that doesn't use `waitFor`

Reply:
Done